save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/HasErrorModal.kt

```diff
@@ -46,18 +47,30 @@ val requestModalHandler: FC<PropsWithChildren> = FC { props ->
                 errorMessage = "You are not logged in",
                 errorLabel = "Unauthenticated",
                 confirmationText = "Proceed to login page",
+                status = response?.status,
+            )
+        } else if (response?.status == 404.toShort()) {
+            ErrorModalState(
+                isErrorModalOpen = false,
+                errorMessage = "${response?.status} ${response?.statusText}",
+                errorLabel = response?.status.toString(),
+                status = response?.status,
             )
         } else {
             ErrorModalState(
                 isErrorModalOpen = response != null,
                 errorMessage = "${response?.status} ${response?.statusText}",
                 errorLabel = response?.status.toString(),
+                status = response?.status,
             )
         }
         setModalState(newModalState)
     }
 
     modal { modalProps ->
+        if (modalState.status == 404.toShort()) {
+            window.location.href = "${window.location.origin}/error"
```

Comment:
We shouldn't redirect to the generic fallback page. We should clearly indicate that it's a 404 Not Found situation

Reply:
Yes, agree